io.c

```diff
@@ -4356,22 +4356,22 @@ rb_io_set_lineno(VALUE io, VALUE lineno)
     return lineno;
 }
 
-/*
- *  call-seq:
- *    readline(sep = $/, chomp: false)   -> string
- *    readline(limit, chomp: false)      -> string
- *    readline(sep, limit, chomp: false) -> string
- *
- *  Reads a line as with IO#gets, but raises EOFError if already at end-of-stream.
- *
- *  Optional keyword argument +chomp+ specifies whether line separators
- *  are to be omitted.
- */
-
+/* :nodoc: */
 static VALUE
-rb_io_readline(int argc, VALUE *argv, VALUE io)
+io_readline(rb_execution_context_t *ec, VALUE io, VALUE sep, VALUE lim, VALUE chomp)
 {
-    VALUE line = rb_io_gets_m(argc, argv, io);
+    VALUE rs = rb_rs;
+    VALUE tmp = Qnil;
+
+    if (NIL_P(sep) || !NIL_P(tmp = rb_check_string_type(sep))) {
+        rs = tmp;
+    }
+    else {
+        lim = sep;
+    }
```

Comment:
Currently, when non-string separator is given with limit, for instance `STDIN.readline(1, 2)`, raises `TypeError`.
This PR changes the behavior  to silently ignore the limit.

Reply:
Ah, oops.  Thanks.  I didn't mean to change that.  I'll fix and add a test.